src/platform_impl/linux/drm/event_loop.rs

```diff
@@ -0,0 +1,1417 @@
+
+pub struct EventLoopWindowTarget<T> {
+    /// drm Connector
+    pub connector: drm::control::connector::Info,
+
+    /// drm crtc
+    pub crtc: drm::control::crtc::Info,
+
+    /// drm mode
+    pub mode: drm::control::Mode,
+
+    /// drm plane
+    pub plane: drm::control::plane::Handle,
+
+    // /// drm dumbbuffer containing the cursor
+    // pub cursor_buffer: drm::control::framebuffer::Handle,
+    /// drm device
+    pub device: Card,
+
+    /// Event loop handle.
+    pub event_loop_handle: calloop::LoopHandle<'static, EventSink>,
+
+    pub(crate) event_sink: EventSink,
+
+    /// A proxy to wake up event loop.
+    pub event_loop_awakener: calloop::ping::Ping,
+
+    _marker: std::marker::PhantomData<T>,
+}
+
+impl<T> EventLoopWindowTarget<T> {
+    #[inline]
+    pub fn primary_monitor(&self) -> Option<crate::monitor::MonitorHandle> {
+        Some(crate::monitor::MonitorHandle {
+            inner: crate::platform_impl::MonitorHandle::Drm(super::MonitorHandle(
+                self.connector.clone(),
+            )),
+        })
+    }
+
+    #[inline]
+    pub fn available_monitors(&self) -> VecDeque<super::MonitorHandle> {
+        self.device
+            .resource_handles()
+            .unwrap()
+            .connectors()
+            .iter()
+            .map(|f| super::MonitorHandle(self.device.get_connector(*f).unwrap()))
+            .collect()
+    }
+}
+
+pub struct EventLoop<T: 'static> {
+    /// Event loop.
+    event_loop: calloop::EventLoop<'static, EventSink>,
+
+    /// Pending user events.
+    pending_user_events: std::rc::Rc<std::cell::RefCell<Vec<T>>>,
+
+    /// Sender of user events.
+    user_events_sender: calloop::channel::Sender<T>,
+
+    /// Window target.
+    window_target: crate::event_loop::EventLoopWindowTarget<T>,
+}
+
+pub(crate) fn find_prop_id<T: ResourceHandle>(
+    card: &Card,
+    handle: T,
+    name: &'static str,
+) -> Option<property::Handle> {
+    let props = card
+        .get_properties(handle)
+        .expect("Could not get props of connector");
+    let (ids, _vals) = props.as_props_and_values();
+    ids.iter()
+        .find(|&id| {
+            let info = card.get_property(*id).unwrap();
+            info.name().to_str().map(|x| x == name).unwrap_or(false)
+        })
+        .cloned()
+}
+
+impl<T: 'static> EventLoop<T> {
+    pub fn new() -> Result<EventLoop<T>, crate::error::OsError> {
+        let mut seat = {
+            let active = Arc::new(AtomicBool::new(false));
+            let t_active = active.clone();
+            let mut s = libseat::Seat::open(
+                move |_, event| {
+                    if let libseat::SeatEvent::Enable = event {
+                        t_active.store(true, std::sync::atomic::Ordering::SeqCst);
+                    }
+                },
+                None,
+            )
+            .map_err(|_| {
+                crate::error::OsError::new(
+                    line!(),
+                    file!(),
```

Comment:
I'm not sure that forwarding line and file is any useful? Well it could be while debugging, but it seems to much and not really sensible, since users won't look into code anyway.